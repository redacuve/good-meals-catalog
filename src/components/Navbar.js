import React, { useRef } from 'react';
import { useHistory, Link } from 'react-router-dom';
import logo from '../assets/img/goodmeals.png';

function Navbar() {
  const inputSearch = useRef(null);
  const { push } = useHistory();
  const onSearch = () => {
    push(`/search/${inputSearch.current.value}`);
    inputSearch.current.value = '';
  };
  return (
    <nav className="w-full" data-testid="navbar">
      <div className="w-full bg-gray-100 border-b border-grey-light">
        <div className="w-full container mx-auto flex flex-wrap items-center justify-center md:justify-between py-4">
          <Link to="/" className="flex items-center">
            <img src={logo} alt="Good Meals Logo" className="h-16 md:h-12" />
            <h1 className="font-bold text-indigo-900 text-base font-extrabold text-xl w-minc ml-1 leading-5" data-testid="title-logo">
              Good Meals Catalog
            </h1>
          </Link>
          <div className="w-full sm:w-2/3 flex flex-wrap justify-center mt-2 lg:justify-end">
            <input
              className="shadow appearance-none border rounded py-2 px-3 text-gray-700 focus:outline-none focus:shadow-outline"
              type="search"
              ref={inputSearch}
              id="seachbar"
              data-testid="searchbar"
              onKeyPress={e => {
                if (e.key === 'Enter') {
                  onSearch();
                }
              }}
            />
            <button
              className="bg-indigo-400 hover:bg-indigo-600 text-white font-bold py-2 px-4 rounded-full"
              type="button"
              onClick={onSearch}
              data-testid="search-button"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </nav>
  );
}

export default Navbar;
