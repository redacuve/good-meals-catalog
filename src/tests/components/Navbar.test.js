import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Navbar from '../../components/Navbar';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('render Navbar icon correctly', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  expect(getByTestId('title-logo')).toContainHTML('Good Meals Catalog');
});

it('render container correctly', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  );
  const child = getByTestId('searchbar');
  expect(getByTestId('navbar')).toContainElement(child);
});

it('matches snapshot 1', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
