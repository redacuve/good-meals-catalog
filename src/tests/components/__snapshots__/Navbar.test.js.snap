// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<nav
  className="w-full"
  data-testid="navbar"
>
  <div
    className="w-full bg-gray-100 border-b border-grey-light"
  >
    <div
      className="w-full container mx-auto flex flex-wrap items-center justify-center md:justify-between py-4"
    >
      <a
        className="flex items-center"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Good Meals Logo"
          className="h-16 md:h-12"
          src="goodmeals.png"
        />
        <h1
          className="font-bold text-indigo-900 text-base font-extrabold text-xl w-minc ml-1 leading-5"
          data-testid="title-logo"
        >
          Good Meals Catalog
        </h1>
      </a>
      <div
        className="w-full sm:w-2/3 flex flex-wrap justify-center mt-2 lg:justify-end"
      >
        <input
          className="shadow appearance-none border rounded py-2 px-3 text-gray-700 focus:outline-none focus:shadow-outline"
          data-testid="searchbar"
          id="seachbar"
          onKeyPress={[Function]}
          type="search"
        />
        <button
          className="bg-indigo-400 hover:bg-indigo-600 text-white font-bold py-2 px-4 rounded-full"
          data-testid="search-button"
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>
</nav>
`;
